tests/cypress/e2e/unit_tests/p1_fleet.spec.ts

```diff
@@ -316,3 +316,71 @@ if (!/\/2\.7/.test(Cypress.env('rancher_version'))) {
+        resourceType: "ConfigMaps",
+        resourceName: "mp-app-config",
+        resourceNamespace: "test-fleet-mp-config",
+      },
+      { qase_id: 79,
+        repoName: "local-cluster-correct-79",
+        resourceType: "Services",
+        resourceName: "mp-app-service",
+        resourceNamespace: "test-fleet-mp-service"
+      },
+    ]
+
+    correctDriftTestData.forEach(
+      ({qase_id, repoName, resourceType, resourceName, resourceNamespace}) => {
+        qase(qase_id,
+          it(`Fleet-${qase_id}: Test MODIFICATION to IMMUTABLE resource ${resourceType} will not be self-healed when correctDrift is set to true in existing GitRepo.`, { tags: `@fleet-${qase_id}` }, () => {
+            const path = "multiple-paths"
+            const portNumber = 6341
+            // Add GitRepo by enabling 'correctDrift'
+            cy.fleetNamespaceToggle('fleet-local')
+            cy.addFleetGitRepo({ repoName, repoUrl, branch, path, correctDrift: 'yes' });
+            cy.clickButton('Create');
+            cy.checkGitRepoStatus(repoName, '2 / 2', '2 / 2');
+
+            if (resourceType === 'ConfigMaps') {
+              // Modify ConfigMap by adding new key-value pair
+              cy.accesMenuSelection('local', 'Storage', 'ConfigMaps');
+              cy.nameSpaceMenuToggle(resourceNamespace);
+              cy.filterInSearchBox(resourceName);
+              cy.get('.col-link-detail').contains(resourceName).should('be.visible');
+              cy.open3dotsMenu(resourceName, 'Edit Config');
+              cy.clickButton('Add');
+              cy.get('#data > .key-value > .kv-container > :nth-child(7) > [data-testid="input-kv-item-key-1"]').focus().type('test_key');
+              cy.get('div.code-mirror.as-text-area').eq(1).click().type("test_data_value");
+              cy.clickButton('Add');
+              cy.wait(500);
+              cy.clickButton('Save');
+
+              cy.filterInSearchBox(resourceName);
+              cy.verifyTableRow(0, resourceName, "test, test_key");
+            }
+            else {
+              cy.accesMenuSelection('local', 'Service Discovery', 'Services');
+              cy.nameSpaceMenuToggle(resourceNamespace);
+              cy.filterInSearchBox(resourceName);
+              cy.get('.col-link-detail').contains(resourceName).should('be.visible');
+              cy.open3dotsMenu(resourceName, 'Edit Config');
+              cy.get("input[type=number]").clear().type(portNumber);
+              cy.wait(500);
+              cy.clickButton('Save');
+
+              cy.filterInSearchBox(resourceName);
+              cy.verifyTableRow(0, resourceName, `${portNumber} `);
+            }
+
+            // Any mutable resource will reconcile to it's original state immediately
+            // But with ConfigMaps and Services it is not because they are immutable i.e.
+            // They didn't reconciled when `correctDrift` is used.
+            cy.deleteAllFleetRepos();
```

Comment:
This can be refactored a bit:

```suggestion
             cy.addFleetGitRepo({ repoName, repoUrl, branch, path, correctDrift: 'yes' });
            cy.clickButton('Create');
            cy.checkGitRepoStatus(repoName, '2 / 2', '2 / 2');
            cy.accesMenuSelection('local', resourceLocation, resourceType);
            cy.nameSpaceMenuToggle(resourceNamespace);
            cy.filterInSearchBox(resourceName);
            cy.get('.col-link-detail').contains(resourceName).should('be.visible');
            cy.open3dotsMenu(resourceName, 'Edit Config');

            if (resourceType === 'ConfigMaps') {
              cy.clickButton('Add');
              cy.get('[data-testid="input-kv-item-key-1"]').eq(0).focus().type('test_key');
              cy.get('div.code-mirror.as-text-area').eq(1).click().type("test_data_value");
              cy.clickButton('Add');
            }
            else if (resourceType === 'Services') {
              cy.get("input[type=number]").clear().type("6341");
            }

            else  {
              throw new Error(`Resource "${resourceType}" is invalid  / not implemented yet`);
            }

            cy.wait(500);
            cy.clickButton('Save');
            cy.filterInSearchBox(resourceName);
            cy.verifyTableRow(0, resourceName, dataToAssert);
            // Any mutable resource will reconcile to it's original state immediately
            // But with ConfigMaps and Services it is not because they are immutable i.e.
            // They didn't reconciled when `correctDrift` is used.
            cy.deleteAllFleetRepos();
```
----
Pls note in the refactor I included things like:

- Locators changes that could be prone to errors like:
```
cy.get('#data > .key-value > .kv-container > :nth-child(7) > [data-testid="input-kv-item-key-1"]').focus().type('test_key');
```
to
```
cy.get('[data-testid="input-kv-item-key-1"]').eq(0).focus().type('test_key')
```
- Changed from `else` to `else if` given that new options may be added
- Added final `else` to catch not allowed resource types

Reply:
Updated with suggested changes.